### ISSUE TITLE
PouchDB is breaking server replication

### ISSUE THREAD

**Issue by yaronyg** [pouchdb/pouchdb#2140]
**tl:dr - PouchDB turns databaseA.replicate.from(databaseB, { server: true}) into the equivalent of databaseB.replicate.to(databaseA, { server: true}). While this translation is mathematically correct it fails in practice due to security. Can we please stop this translation when dealing with server side replication requests?**
# Setup:

databaseA = new PouchDB("http://foo.com/dba");
databaseB = new PouchDB("http://bar.com/dbb");
databaseA.replicate.from(databaseB, { server: true});
# Expected Result:

Server request is sent to: http://foo.com/_replicate
Request body:
- Source: http://bar.com/dbb
- Target: dba
# Actual Result:

Server request is sent to: http://bar.com/_replicate
Request body:
- Source: dbb
- Target: http://foo.com/dba
# Who Cares?!?!

What's the problem, right? After all isn't databaseA.replicate.from(databaseB, { server: true}) == databaseB.replicate.to(databaseA, {server: true})?!?!? Mathematically the answer is YES! But in practice the answer is no.

The problem is security. 

In our scenario databaseA is User A's personal couchDB server. User A wants to synch feed updates from his buddy User B's personal couchDB server, databaseB.

First off, because of our architecture, User A's PouchDB client would use credential A1 to send requests to databaseA. DatabaseA then has its own credentials, A2, which have been provisioned with DatabaseB and will allow requests. Therefore right off the bat PouchDB's attempt to send a _replicate request to DatabaseB will fail because PouchDB has credentials A1 NOT A2.

But let's pretend that User A puts credentials A2 on his client (this would actually be a monstrous security hole in itself but I just want to drive the point home). The attempt to send a replicate request to databaseB would still fail! The reason is that User B only gave User A read only access to databaseB. This would not include issuing a _replicate request on the server's root! It would only include, for example, calling dbb/_changes.

So, either way, changing a "from" to a "to" on a server side replication doesn't work the second security is involved and in P2P (what I'm working on) security is the name of the game.
# So now what?

If y'all are o.k. with actually fixing this (please!) I'm happy to submit a PR. It seems pretty straight forward. We just need to check who made the request and if it's server side replication make sure we issue the request to the server the request was submitted on.
